lib/logstash/filters/mutate.rb

```diff
@@ -288,6 +291,17 @@ def convert_float(value)
     return value.to_f
   end # def convert_float
 
+  def convert_boolean(value)
+    if value == "true"
+      return true
+    elsif value == "false"
+      return false
+    else
+      @logger.warn("Failed to convert #{value} into boolean. Acceptable values are \"true\" and \"false\".")
+      return value
```

Comment:
@untergeek usually, `filter_matched(event)` is called when the filter was successful. A [good example of this is grok](https://github.com/logstash-plugins/logstash-filter-grok/blob/master/lib/logstash/filters/grok.rb#L287-L295).
I though that this was also the behaviour of the mutate filter but apparently not. Let's merge this and open another issue to discuss if mutate should not apply filter_matched on failed mutations.